### PR TITLE
[ci] JDK matrix Buildkite pipelines (pt 2/Windows)

### PR DESCRIPTION
**Pull request by dliappis**
## Release notes
[rn:skip]

## What does this PR do?

This commit adds JDK matrix Buildkite pipelines for Windows 2022, 2019 and 2016.

It also makes the groups easier to read (on both Linux and Windows pipelines) by removing the os-jdk prefix from the job labels.

`testDLQWriterFlusherRemovesExpiredSegmentWhenCurrentHeadSegmentIsEmpty` fails on Windows Buildkite agents and it's a test issue tracked in https://github.com/elastic/logstash/issues/15562.

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] ~I have made corresponding changes to the documentation~
- [ ] ~I have made corresponding change to the default configuration files (and/or docker env variables)~
- [x] I have added tests that prove my fix is effective or that my feature works:

#### Linux improved group views ([link](https://buildkite.com/elastic/logstash-linux-jdk-matrix-pipeline/builds/17)):

<details><summary>Screenshot</summary>

image

image

</details>

####  Windows run ([link](https://buildkite.com/elastic/logstash-windows-jdk-matrix-pipeline/builds/27)):

This run demonstrates two failures (Windows 2022 + 2019) due to https://github.com/elastic/logstash/issues/15562 , as well a success (Windows 2019).

<details><summary>Screenshot</summary>

image
</details>

Note that we also collect [archived build artifacts](https://buildkite.com/elastic/logstash-windows-jdk-matrix-pipeline/builds/27#018bb941-224f-46da-8107-15fe00e99fb2) to assist with troubleshooting:

<details><summary>Screenshot</summary>

image
</details>

## How to test this PR locally

Just kick off a build using https://buildkite.com/elastic/logstash-windows-jdk-matrix-pipeline#new and defining this PR i.e. `pull/15563/merge` in the `branch` field.

## Related issues

- https://github.com/elastic/logstash/pull/15539
- https://github.com/elastic/ingest-dev/issues/1725
- https://github.com/elastic/logstash/issues/15562

### CODE REVIEW

File: .buildkite/scripts/jdk-matrix-tests/generate-steps.py [hunk 1]
```diff
@@ -81,21 +83,24 @@ def __init__(self, os: str, jdk: str, group_key: str):
...
     def all_jobs(self) -> list[typing.Callable[[], JobRetValues]]:
         return [
+          self.init_annotation,
           self.unit_tests,
         ]
 
     def unit_tests(self) -> JobRetValues:
-        step_name_human = "Java Unit Test"
-        test_command = "# TODO"
+        step_name_human = "Unit Test"
```
Comment (roaksoax) [inline .buildkite/scripts/jdk-matrix-tests/generate-steps.py:91]: The only neat here is to maybe do "Unit Tests (Java/Ruby)"

**Review by roaksoax — APPROVED** [review]
lgtm! just one nit inline.

File: .buildkite/scripts/jdk-matrix-tests/generate-steps.py (hunk 1 above)
Comment (dliappis) [reply .buildkite/scripts/jdk-matrix-tests/generate-steps.py]: Great idea. Added it in https://github.com/elastic/logstash/pull/15563/commits/491fed51ecb017b7b47460045f3078f393abcb02

Comment (dliappis) [reply .buildkite/scripts/jdk-matrix-tests/generate-steps.py]: raised https://github.com/elastic/logstash/issues/15566 to track the improvement

**Comment by dliappis** [comment]
@logstashmachine backport 8.11

**Comment by dliappis** [comment]
@logstashmachine backport 8.10

**Comment by dliappis** [comment]
@logstashmachine backport 7.17

**Pull request merged**